From bfaa83ffd94414c6568ac988a23c0ff7f5541261 Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Mon, 25 Nov 2024 16:19:55 +0000
Subject: [PATCH] misc/rp1-pio: Fix copy/paste error in pio_rp1.h

As per the subject, there was a copy/paste error that caused
pio_sm_unclaim from a driver to result in a call to
pio_sm_claim. Fix it.

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 include/linux/pio_rp1.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/include/linux/pio_rp1.h
+++ b/include/linux/pio_rp1.h
@@ -320,7 +320,7 @@ static inline int pio_sm_unclaim(struct
 	if (bad_params_if(client, sm >= NUM_PIO_STATE_MACHINES))
 		return -EINVAL;
 
-	return rp1_pio_sm_claim(client, &args);
+	return rp1_pio_sm_unclaim(client, &args);
 }
 
 static inline int pio_claim_unused_sm(struct rp1_pio_client *client, bool required)
